File: app/src/main/java/com/fiap/inovagab/data/model/Projeto.kt
package com.fiap.inovagab.data.model

enum class StatusProjeto {
    PLANEJADO,
    EM_ANDAMENTO,
    CONCLUIDO,
    CANCELADO
}

data class Projeto(
    val id: String = "",
    val nome: String = "",
    val descricao: String = "",
    val ideiaId: String = "",
    val estrategiaId: String = "",
    val responsavel: String = "",
    val responsavelId: String = "",
    val etapa: String = "",
    val status: StatusProjeto = StatusProjeto.PLANEJADO,
    val versao: Int = 1,
    val investimento: Double = 0.0,
    val retornoFinanceiro: Double = 0.0,
    val reducaoCustos: Double = 0.0,
    val ganhoProdutividade: Double = 0.0,
    val prazo: String = "",
    val criadoEm: Long = System.currentTimeMillis()
)

File: backend/src/InovaGAB.Domain/Projetos/Projeto.cs
namespace InovaGAB.Domain.Projetos;

public sealed class Projeto
{
    public string Id { get; set; } = string.Empty;

    public string Nome { get; set; } = string.Empty;

    public string Descricao { get; set; } = string.Empty;

    public string? IdeiaId { get; set; }

    public string EstrategiaId { get; set; } = string.Empty;

    public int EstrategiaVersao { get; set; }

    public string ResponsavelId { get; set; } = string.Empty;

    public string Etapa { get; set; } = string.Empty;

    public StatusProjeto Status { get; set; } = StatusProjeto.PLANEJADO;

    public decimal Investimento { get; set; }

    public decimal RetornoFinanceiro { get; set; }

    public decimal ReducaoCustos { get; set; }

    public decimal GanhoProdutividade { get; set; }

    public DateOnly Prazo { get; set; }

    public int Versao { get; set; } = 1;

    public DateTime? ExcluidaEmUtc { get; set; }

    public DateTime CriadoEmUtc { get; set; }

    public DateTime AtualizadoEmUtc { get; set; }

    public bool Demo { get; set; }
}
